### PR TITLE
feat(card): added onClick handler

### PR DESCRIPTION
**Pull request by SamCreasey**
Added an onClick handler to the card component, and added button/focus styles for accessibility when the onClick prop is supplied. 
Updated tests to cover new prop

### CODE REVIEW

**Review by hochreutenerl — APPROVED** [review]
Looks good code-wise and works as intended. Just some smaller nitpicks 😃

File: src/components/Card/Card.spec.tsx [hunk 1]
```diff
@@ -3,14 +3,42 @@
...
     it("should render children correctly", () => {
         mount(<Card>{CARD_CHILDREN}</Card>);
 
-        cy.get("[data-test-id=card]").as("Card");
-        cy.get("@Card").contains("child");
+        cy.get(CARD_ID).contains("child");
+    });
+
+    it("should call onClick", () => {
+        const stubbedOnClick = cy.stub();
```
[the remark this answers is not in the document]
Comment (SamCreasey) [reply src/components/Card/Card.spec.tsx:20]: updated

File: src/components/Card/Card.spec.tsx [hunk 2]
```diff
@@ -3,14 +3,42 @@
...
+        mount(<Card onClick={stubbedOnClick}>{CARD_CHILDREN}</Card>);
+
+        cy.get(CARD_ID).click();
+        cy.get(CARD_ID).invoke("attr", "tabIndex").should("equal", "0");
+        cy.wrap(stubbedOnClick).should("have.been.called");
+    });
+
+    it("should be accessible", () => {
+        const stubbedOnClick = cy.stub();
```
[the remark this answers is not in the document]
Comment (SamCreasey) [reply src/components/Card/Card.spec.tsx:29]: updated

File: src/components/Card/Card.stories.tsx [hunk 3]
```diff
@@ -94,3 +94,13 @@ export const MultipleCards: Story<CardProps> = () => (
         ))}
     </div>
 );
+
+export const WithOnClick = Template.bind({});
+
+WithOnClick.args = {
+    children: "I'm clickable!",
```
[the remark this answers is not in the document]
Comment (SamCreasey) [reply src/components/Card/Card.stories.tsx:101]: updated :)

**Pull request merged**